[Export] Fix SDPA decomposition

sidt-meta opened the pull request:
Summary: Update SDPA decomposition to match updated stride from D62009189 which aligns strides with the `aten._scaled_dot_product_attention_math.default`, which makes `t.permute().continuous().permute()` no longer necessary.

Test Plan: CI

Differential Revision: D62278378

StellarrZ commented:
@Valentine233 @drisspg @jgong5 Could you please help review? The current decomposition seems broken. Thanks!
cc @jfix71 @SherlockNoMad @tugsbayasgalan

drisspg reviewed (commented):
do you have a tests for this? I am not sure how this is different than the existing impl

sidt-meta commented:
Notebook [N5855526](https://www.internalfb.com/intern/anp/view/?id=5855526) attempts to reproduce the same ops that are generated when we export the model we're working on.

While we don't have an ongoing test, this is what we did to validate the issue & fix:
1. We reproduced the export failure for the model after PR #134656 landed (and that the failure went away after we backed-out the diff)
2. Export worked again after the change in this PR was applied.

drisspg commented:
cc @kimishpatel

sidt-meta commented:
Updated exsting test_sdpa in test_decomp.py with to make sure the output's stride is as expected

drisspg reviewed (approved):
Looks good

sidt-meta committed:
[Export] Fix SDPA decomposition (#135297)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/135297

Updated stride from D62009189 which changes the output tensor stride flash attention. The decomposition needs to be updated have an output tensor with the same shape/stride in order for the subsequent view to be valid during torch.export

Test Plan:
Set up buck target for test_decomp.py

```
buck2 test mode/{opt,inplace} //caffe2/test:test_decomp -- test_sdpa
```
 {F1857936880}

----
CI signals

Reviewed By: StellarrZ

Differential Revision: D62278378
